# Asthma Prevalence / Vulnerability

## Cleaning and wrangling the data

The final product of this file is data.csv, which is the main data file used in the subsequent machine learning models.

In [1]:
# inspect dataset

In [2]:
!pip install geopandas
!pip install shapely

In [3]:
import pandas as pd
# pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)
import geopandas as gpd
from shapely.geometry import Point
from shapely.validation import make_valid
import numpy as np
import matplotlib.pyplot as plt

### Inspecting Allegheny County census tract file for future validation purposes.

Data taken from WPRDC should show all Census Tracts in year 2016. Result: there should be 402 rows per year in our final data set.

In [4]:
alCoCTs = pd.read_csv('our-data/AlleghenyCountyCensusTracts2016.csv')
print('Number of Census Tracts: ', len(alCoCTs))
alCoCTs.sort_values('geoid')

Number of Census Tracts:  402


_id  fid  state_fp  county_fp  tract_ce              affgeoid  \
69   15019   70        42          3     10300  1400000US42003010300   
368  15318  369        42          3     20100  1400000US42003020100   
28   14978   29        42          3     20300  1400000US42003020300   
70   15020   71        42          3     30500  1400000US42003030500   
394  15344  395        42          3     40200  1400000US42003040200   
..     ...  ...       ...        ...       ...                   ...   
212  15162  213        42          3    981000  1400000US42003981000   
80   15030   81        42          3    981100  1400000US42003981100   
68   15018   69        42          3    981200  1400000US42003981200   
165  15115  166        42          3    981800  1400000US42003981800   
213  15163  214        42          3    982200  1400000US42003982200   

           geoid    name lsad   a_land  a_water  shape_length  shape_area  
69   42003010300   103.0   CT   866087   235982      0.055235    0.000117  
368  42003020100   201.0   CT  1678102   483177      0.083161    0.000231  
28   42003020300   203.0   CT  1594790   326537      0.077032    0.000204  
70   42003030500   305.0   CT   668309        0      0.046956    0.000072  
394  42003040200   402.0   CT   604717        0      0.049916    0.000064  
..           ...     ...  ...      ...      ...           ...         ...  
212  42003981000  9810.0   CT  1372534        0      0.063284    0.000147  
80   42003981100  9811.0   CT   875989        0      0.043523    0.000093  
68   42003981200  9812.0   CT   460275   156913      0.037880    0.000067  
165  42003981800  9818.0   CT  1822889   310071      0.066922    0.000228  
213  42003982200  9822.0   CT   505672        0      0.047812    0.000052  

[402 rows x 13 columns]

In [5]:
alCoCTs['geoid'].nunique()

402

In [6]:
alCoBGs = pd.read_csv('our-data/AlleghenyCountyCensusBlockGroups2016.csv')
print('Number of Block Groups: ', len(alCoBGs))
alCoBGs.sort_values('geoid')

Number of Block Groups:  1100


_id  fid  state_fp  county_fp  tract_ce  blkgrp_ce         geoid  \
397  55277  398        42          3     10300          1  420030103001   
55   54935   56        42          3     10300          2  420030103002   
775  55655  776        42          3     10300          3  420030103003   
178  55058  179        42          3     10300          4  420030103004   
398  55278  399        42          3     20100          1  420030201001   
..     ...  ...       ...        ...       ...        ...           ...   
168  55048  169        42          3    981000          1  420039810001   
206  55086  207        42          3    981100          1  420039811001   
748  55628  749        42          3    981200          1  420039812001   
139  55019  140        42          3    981800          1  420039818001   
244  55124  245        42          3    982200          1  420039822001   

          namelsad  mtfcc funcstat    aland  awater   intptlat   intptlon  \
397  Block Group 1  G5030        S   291362  235982  40.434050 -79.986430   
55   Block Group 2  G5030        S   275341       0  40.436894 -79.990472   
775  Block Group 3  G5030        S   168360       0  40.436893 -79.983761   
178  Block Group 4  G5030        S   131024       0  40.437347 -79.977287   
398  Block Group 1  G5030        S   435122  249581  40.439695 -80.006178   
..             ...    ...      ...      ...     ...        ...        ...   
168  Block Group 1  G5030        S  1372534       0  40.473101 -79.948865   
206  Block Group 1  G5030        S   875989       0  40.442038 -79.906326   
748  Block Group 1  G5030        S   460275  156913  40.446463 -80.011958   
139  Block Group 1  G5030        S  1822889  310071  40.477494 -79.896498   
244  Block Group 1  G5030        S   505672       0  40.444488 -79.954765   

     shape_length  shape_area  
397      0.052197    0.000056  
55       0.022431    0.000029  
775      0.020349    0.000018  
178      0.020653    0.000014  
398      0.046121    0.000073  
..            ...         ...  
168      0.066651    0.000146  
206      0.046587    0.000093  
748      0.038948    0.000066  
139      0.068401    0.000227  
244      0.051520    0.000054  

[1100 rows x 16 columns]

In [7]:
alCoBGs['geoid'].nunique()

1100

### Prevalence Data

In [8]:
#Loading in 2020 data to fill in missing rows and filtering on Allegheny County values
placesData = pd.read_csv('our-data/PLACES__Census_Tract_Data__GIS_Friendly_Format___2020_release_20250417.csv')
placesData = placesData[placesData['CountyName']=='Allegheny']

#Loading in data for 2019
cities2019 = pd.read_csv("our-data/500_Cities__Census_Tract-level_Data__GIS_Friendly_Format___2019_release_20250405.csv")
#Filtering for Pittsburgh census tracts
cities2019=cities2019[cities2019['PlaceName'] == 'Pittsburgh']
#Removing values from the 2020 data set that are already in the 2019 dataset
places2019 = placesData[placesData['TractFIPS'].apply(lambda x: x not in cities2019['TractFIPS'].to_list())]
#Joining to represent all Census Tracts in Allegheny County and adding column 'Year' indicating time period that each row represents
prevn2019 = pd.concat([cities2019, places2019])
prevn2019['Year'] = ['2019']*len(prevn2019)

#Above process is repeated for the remaining years
cities2018 = pd.read_csv("our-data/500_Cities__Census_Tract-level_Data__GIS_Friendly_Format___2018_release_20250416.csv")
cities2018=cities2018[cities2018['PlaceName'] == 'Pittsburgh']
places2018= placesData[placesData['TractFIPS'].apply(lambda x: x not in cities2018['TractFIPS'].to_list())]
prevn2018 = pd.concat([cities2018, places2018])
prevn2018['Year'] = ['2018']*len(prevn2018)

cities2017 = pd.read_csv("our-data/500_Cities__Census_Tract-level_Data__GIS_Friendly_Format___2017_release_20250416.csv")
cities2017=cities2017[cities2017['PlaceName'] == 'Pittsburgh']
places2017= placesData[placesData['TractFIPS'].apply(lambda x: x not in cities2017['TractFIPS'].to_list())]
prevn2017 = pd.concat([cities2017, places2017])
prevn2017['Year'] = ['2017']*len(prevn2017)

cities2016 = pd.read_csv("our-data/500_Cities__Census_Tract-level_Data__GIS_Friendly_Format___2016_release_20250416.csv")
cities2016=cities2016[cities2016['PlaceName'] == 'Pittsburgh']
places2016= placesData[placesData['TractFIPS'].apply(lambda x: x not in cities2016['TractFIPS'].to_list())]
prevn2016 = pd.concat([cities2016, places2016])
prevn2016['Year'] = ['2016']*len(prevn2016)

#Data for each year is joined together after every year is completed
cond_prevn = pd.concat([prevn2019, prevn2018, prevn2017, prevn2016])
cond_prevn

StateAbbr   PlaceName  PlaceFIPS    TractFIPS        Place_TractID  \
450          PA  Pittsburgh  4261000.0  42003010300  4261000-42003010300   
451          PA  Pittsburgh  4261000.0  42003020100  4261000-42003020100   
452          PA  Pittsburgh  4261000.0  42003020300  4261000-42003020300   
453          PA  Pittsburgh  4261000.0  42003030500  4261000-42003030500   
454          PA  Pittsburgh  4261000.0  42003040200  4261000-42003040200   
...         ...         ...        ...          ...                  ...   
55202        PA         NaN        NaN  42003564000                  NaN   
55203        PA         NaN        NaN  42003564100                  NaN   
55204        PA         NaN        NaN  42003564200                  NaN   
55205        PA         NaN        NaN  42003564400                  NaN   
55206        PA         NaN        NaN  42003564500                  NaN   

       Population2010  ACCESS2_CrudePrev ACCESS2_Crude95CI  \
450            6600.0               12.9      (10.2, 16.9)   
451            3629.0                6.7      ( 5.5,  8.5)   
452             616.0                4.8      ( 4.1,  5.8)   
453            2256.0               14.4      (11.9, 17.2)   
454            2604.0               11.8      ( 8.8, 15.8)   
...               ...                ...               ...   
55202             NaN                8.3      ( 7.5,  9.1)   
55203             NaN                4.3      ( 3.6,  5.2)   
55204             NaN                7.6      ( 6.5,  8.8)   
55205             NaN                8.7      ( 7.2, 10.3)   
55206             NaN                8.3      ( 7.2,  9.5)   

       ARTHRITIS_CrudePrev ARTHRITIS_Crude95CI  BINGE_CrudePrev  \
450                   11.5        (10.9, 12.3)             27.4   
451                   17.7        (17.0, 18.4)             25.7   
452                   16.8        (16.2, 17.5)             27.2   
453                   35.6        (34.1, 37.0)             13.5   
454                   16.5        (15.7, 17.4)             21.4   
...                    ...                 ...              ...   
55202                 26.5        (25.9, 27.2)             22.3   
55203                 27.9        (26.8, 29.1)             20.2   
55204                 31.7        (30.6, 32.8)             19.6   
55205                 31.1        (30.0, 32.0)             20.0   
55206                 30.3        (29.4, 31.2)             20.5   

      BINGE_Crude95CI  BPHIGH_CrudePrev BPHIGH_Crude95CI  BPMED_CrudePrev  \
450      (25.7, 29.2)              19.0     (18.0, 19.9)             51.9   
451      (24.9, 26.6)              22.1     (21.4, 22.8)             71.8   
452      (26.5, 28.0)              19.7     (19.1, 20.4)             67.6   
453      (13.0, 14.0)              48.3     (47.0, 49.5)             84.3   
454      (19.9, 23.0)              23.3     (22.2, 24.4)             67.1   
...               ...               ...              ...              ...   
55202    (21.9, 22.6)              28.1     (27.6, 28.7)             74.5   
55203    (19.7, 20.6)              28.8     (27.8, 30.0)             79.0   
55204    (19.2, 20.2)              33.4     (32.1, 34.5)             79.7   
55205    (19.4, 20.6)              32.8     (31.4, 34.0)             78.5   
55206    (20.1, 21.0)              31.0     (30.1, 31.8)             77.8   

      BPMED_Crude95CI  CANCER_CrudePrev CANCER_Crude95CI  CASTHMA_CrudePrev  \
450      (50.2, 53.6)               1.8     ( 1.7,  1.9)               11.1   
451      (70.5, 73.0)               4.5     ( 4.4,  4.7)                9.3   
452      (66.4, 68.5)               4.5     ( 4.3,  4.6)                8.2   
453      (83.6, 85.0)               6.9     ( 6.4,  7.1)               12.8   
454      (65.3, 69.2)               3.2     ( 3.0,  3.3)               11.7   
...               ...               ...              ...                ...   
55202    (74.0, 75.0)               6.9     ( 6.8,  7.0)         

In [9]:
print("Unique tract IDs: " + str(cond_prevn['TractFIPS'].nunique()))
print("Shape: " + str(cond_prevn.shape))

Unique tract IDs: 392
Shape: (1568, 71)


We are missing data for 10 tracts for all years (final row count is 392 * 4 = 1568) there 10 tracts are uninhabited (i.e. Parks, Industrial Zones, etc.)

In [10]:
cond_prevn = cond_prevn.rename(columns={'TractFIPS': 'Geo_FIPS'})
#Removing unrelated prevalence columns (not comorbidities with Asthma)
def drop_columns_with_keyword(df, keyword):
    # Identify columns containing the keyword (95CI)
    cols_to_drop = [col for col in df.columns if keyword.lower() in col.lower()]
    # Drop them and return the updated DataFrame
    return df.drop(columns=cols_to_drop)
cond_prevn = drop_columns_with_keyword(cond_prevn,'95CI')
cond_prevn = cond_prevn.drop([
    'ARTHRITIS_CrudePrev',
    'BINGE_CrudePrev',
    'CANCER_CrudePrev',
    'COLON_SCREEN_CrudePrev',
    'DENTAL_CrudePrev',
    'KIDNEY_CrudePrev',
    'LPA_CrudePrev',
    'MAMMOUSE_CrudePrev',
    'MHLTH_CrudePrev',
    'PAPTEST_CrudePrev',
    'PHLTH_CrudePrev',
    'SLEEP_CrudePrev',
    'STROKE_CrudePrev',
    'TEETHLOST_CrudePrev'], axis=1)
cond_prevn['lat'] = cond_prevn['Geolocation'].str[1:15]
cond_prevn['lon'] = cond_prevn['Geolocation'].str[-15:-1]
cond_prevn

StateAbbr   PlaceName  PlaceFIPS     Geo_FIPS        Place_TractID  \
450          PA  Pittsburgh  4261000.0  42003010300  4261000-42003010300   
451          PA  Pittsburgh  4261000.0  42003020100  4261000-42003020100   
452          PA  Pittsburgh  4261000.0  42003020300  4261000-42003020300   
453          PA  Pittsburgh  4261000.0  42003030500  4261000-42003030500   
454          PA  Pittsburgh  4261000.0  42003040200  4261000-42003040200   
...         ...         ...        ...          ...                  ...   
55202        PA         NaN        NaN  42003564000                  NaN   
55203        PA         NaN        NaN  42003564100                  NaN   
55204        PA         NaN        NaN  42003564200                  NaN   
55205        PA         NaN        NaN  42003564400                  NaN   
55206        PA         NaN        NaN  42003564500                  NaN   

       Population2010  ACCESS2_CrudePrev  BPHIGH_CrudePrev  BPMED_CrudePrev  \
450            6600.0               12.9              19.0             51.9   
451            3629.0                6.7              22.1             71.8   
452             616.0                4.8              19.7             67.6   
453            2256.0               14.4              48.3             84.3   
454            2604.0               11.8              23.3             67.1   
...               ...                ...               ...              ...   
55202             NaN                8.3              28.1             74.5   
55203             NaN                4.3              28.8             79.0   
55204             NaN                7.6              33.4             79.7   
55205             NaN                8.7              32.8             78.5   
55206             NaN                8.3              31.0             77.8   

       CASTHMA_CrudePrev  CHD_CrudePrev  CHECKUP_CrudePrev  \
450                 11.1            2.6               63.7   
451                  9.3            3.9               68.4   
452                  8.2            2.8               67.4   
453                 12.8            9.9               79.2   
454                 11.7            4.0               67.5   
...                  ...            ...                ...   
55202                9.9            5.6               77.1   
55203                8.7            5.4               81.1   
55204                9.6            7.2               80.0   
55205               10.1            7.2               79.3   
55206               10.0            7.1               78.8   

       CHOLSCREEN_CrudePrev  COPD_CrudePrev  COREM_CrudePrev  COREW_CrudePrev  \
450                    58.1             4.4             38.8             30.5   
451                    73.4             3.8             50.2             40.9   
452                    80.4             2.8             54.3             46.1   
453                    79.7            10.6             32.5             28.4   
454                    61.4             5.5             31.8             28.0   
...                     ...             ...              ...              ...   
55202                  85.8             6.4             43.9             33.8   
55203                  91.1             4.7             50.8             41.1   
55204                  88.0             7.4             43.7             34.6   
55205                  86.8             7.7             42.5             32.5   
55206                  86.6             7.5             42.7             32.3   

       CSMOKING_CrudePrev  DIABETES_CrudePrev  HIGHCHOL_CrudePrev  \
450                  26.5                 5.0                17.6   
451                  14.4                 6.4                24.9   
452                  12.3                 4.9                24.2   
453                  26.9                20.5                37.2   
454                  22.3                 8.0                22.3   
...         

In [11]:
print(cond_prevn.shape)
cond_prevn.isna().sum()

(1568, 30)


StateAbbr                  0
PlaceName               1060
PlaceFIPS               1060
Geo_FIPS                   0
Place_TractID           1060
Population2010          1187
ACCESS2_CrudePrev          0
BPHIGH_CrudePrev           0
BPMED_CrudePrev            0
CASTHMA_CrudePrev          0
CHD_CrudePrev              0
CHECKUP_CrudePrev          0
CHOLSCREEN_CrudePrev       0
COPD_CrudePrev             0
COREM_CrudePrev            0
COREW_CrudePrev            8
CSMOKING_CrudePrev         0
DIABETES_CrudePrev         0
HIGHCHOL_CrudePrev         0
OBESITY_CrudePrev          0
Geolocation                0
StateDesc                508
CountyName               508
CountyFIPS               508
TotalPopulation          508
CERVICAL_CrudePrev       508
Year                       0
population_count        1441
lat                        0
lon                        0
dtype: int64

After dropping non-relevant conditions, we have 1568 rows and 30 columns. However, there are still columns with a significant amount of missing values. This is due to the 2020 and 2016-19 datasets having slight differences in the columns they contained. Thus, we will drop these columns as they are not relevant to our analysis.

In [12]:
#Renaming dataframe to be more descriptive
prevalence_data = cond_prevn

#Dropping columns containing NaNs
prevalence_data = prevalence_data.drop(columns=["COREW_CrudePrev", 'StateAbbr', 'PlaceName',
                                                'PlaceFIPS', 'Place_TractID', 'StateDesc', 'CountyName',
                                                'CountyFIPS', 'TotalPopulation', 'CERVICAL_CrudePrev',
                                                'Population2010', 'population_count'])

# Keeping only census tract level identifier for Geo_FIPS to facilitate later joins
prevalence_data['Geo_FIPS'] = prevalence_data['Geo_FIPS'].apply(lambda x: int(str(x)[-6:]))

#Dataframe is now free of missing values
prevalence_data.isna().sum()

Geo_FIPS                0
ACCESS2_CrudePrev       0
BPHIGH_CrudePrev        0
BPMED_CrudePrev         0
CASTHMA_CrudePrev       0
CHD_CrudePrev           0
CHECKUP_CrudePrev       0
CHOLSCREEN_CrudePrev    0
COPD_CrudePrev          0
COREM_CrudePrev         0
CSMOKING_CrudePrev      0
DIABETES_CrudePrev      0
HIGHCHOL_CrudePrev      0
OBESITY_CrudePrev       0
Geolocation             0
Year                    0
lat                     0
lon                     0
dtype: int64

### Utilization Data

In [13]:
ER_use = pd.read_csv("our-data/Allegheny_county_ER_use.csv")
ER_use

_id      Geo_FIPS  StateFIPS  CountyFIPS  TractFIPS  BlockgroupFIPS  \
0        1  420031301002         42           3     130100               2   
1        2  420031301002         42           3     130100               2   
2        3  420031301002         42           3     130100               2   
3        4  420031301002         42           3     130100               2   
4        5  420031301003         42           3     130100               3   
...    ...           ...        ...         ...        ...             ...   
4365  4366  420039818001         42           3     981800               1   
4366  4367  420039822001         42           3     982200               1   
4367  4368  420039822001         42           3     982200               1   
4368  4369  420039822001         42           3     982200               1   
4369  4370  420039822001         42           3     982200               1   

      TotalPopEst2015_19ACS  YearOfStartDate  UnderAge1PopEst2015_19ACS  \
0                       518           2016.0                        9.4   
1                       518           2017.0                        9.4   
2                       518           2018.0                        9.4   
3                       518           2019.0                        9.4   
4                       489           2016.0                        0.0   
...                     ...              ...                        ...   
4365                    207           2019.0                        0.0   
4366                   4618           2016.0                        0.0   
4367                   4618           2017.0                        0.0   
4368                   4618           2018.0                        0.0   
4369                   4618           2019.0                        0.0   

      Age0to17PopEst2015_19ACS  Unique0to17WithED_Visit  \
0                          207                     70.0   
1                          207                     69.0   
2                          207                     66.0   
3                          207                     66.0   
4                          134                     38.0   
...                        ...                      ...   
4365                        47                      4.0   
4366                        80                      6.0   
4367                        80                     10.0   
4368                        80                      7.0   
4369                        80                      8.0   

      Unique0to17WithED_VisitPer100  NumberED_VisitsAge0to17  \
0                              33.8                    122.0   
1                              33.3                    130.0   
2                              31.9                    138.0   
3                              31.9                    148.0   
4                              28.4                     71.0   
...                             ...                      ...   
4365                            8.5                      8.0   
4366                            7.5                     10.0   
4367                           12.5                     10.0   
4368                            8.8                      9.0   
4369                           10.0                     13.0   

      ED_VisitsPer100Age0to17  NumberLowAcuityED_VisitsAge0to17  \
0                        58.9                              62.0   
1                        62.8                              80.0   
2                        66.7                              85.0   
3                        71.5                              96.0   
4                        53.0                              39.0   
...                       ...                               ...   
4365                     17.0                               2.0   
4366                     12.5                               2.0   
4367                     12.5                               2.0   
4368                 

In [14]:
ER_use.shape

(4370, 23)

In [15]:
PC_use = pd.read_csv("our-data/Allegheny_county_PC_use.csv")
# PC_use = PC_use.fillna(0)
PC_use

_id      Geo_FIPS  StateFIPS  CountyFIPS  TractFIPS  BlockgroupFIPS  \
0        1  420031301002         42           3     130100               2   
1        2  420031301002         42           3     130100               2   
2        3  420031301002         42           3     130100               2   
3        4  420031301002         42           3     130100               2   
4        5  420031301003         42           3     130100               3   
...    ...           ...        ...         ...        ...             ...   
4361  4362  420039818001         42           3     981800               1   
4362  4363  420039822001         42           3     982200               1   
4363  4364  420039822001         42           3     982200               1   
4364  4365  420039822001         42           3     982200               1   
4365  4366  420039822001         42           3     982200               1   

      TotalPopEst2015_19ACS  Age0to17PopEst2015_19ACS  YearOfContactDate  \
0                       518                       207               2016   
1                       518                       207               2017   
2                       518                       207               2018   
3                       518                       207               2019   
4                       489                       134               2016   
...                     ...                       ...                ...   
4361                    207                        47               2019   
4362                   4618                        80               2016   
4363                   4618                        80               2017   
4364                   4618                        80               2018   
4365                   4618                        80               2019   

      AllPrimaryCarePatientsAge0to17  WellChildVisitsInPastYearAge0to17  \
0                                 82                                 59   
1                                 94                                 58   
2                                 89                                 63   
3                                 86                                 61   
4                                 53                                 34   
...                              ...                                ...   
4361                               4                                  3   
4362                              16                                 13   
4363                              18                                 16   
4364                              19                                 14   
4365                              16                                  8   

      WellChildVisitsInPastYearPer100PrimaryCarePatients  \
0                                                  72.0    
1                                                  61.7    
2                                                  70.8    
3                                                  70.9    
4                                                  64.2    
...                                                 ...    
4361                                               75.0    
4362                                               81.3    
4363                                               88.9    
4364                                               73.7    
4365                                               50.0    

      AsthmaDiagnosisAge0to17  \
0                        17.0   
1                        19.0   
2                        19.0   
3                        17.0   
4                        18.0   
...                       ...   
4361                      2.0   
4362                      1.0   
4363                      1.0   
4364                      1.0   
4365                      1.0   

      AsthmaDiagnosisPer100PrimaryCarePatientsAge0to17  
0                                                 20.7  
1                       

In [16]:
PC_use.shape

(4366, 14)

`ER_use` has 4370 rows and 23 columns

`PC_use` has 4366 rows and 14 columns

Neither of these are divisible by 4, so we are missing some years for some tracts/blockgroups

In [17]:
ER_use['TractFIPS'].nunique() # 402 tracts in ER data
PC_use['TractFIPS'].nunique() # 397 tracts in PC data

# Get number of occurrences of each tract + blockgroup combination; we want to see each combination appear 4 times (for the four years 2016-19)
ER_combo_counts = ER_use.groupby(['TractFIPS', 'BlockgroupFIPS']).size().reset_index(name = 'Count').sort_values('Count', ascending = False)
ER_combo_counts.tail(20)

PC_combo_counts = PC_use.groupby(['TractFIPS', 'BlockgroupFIPS']).size().reset_index(name = 'Count').sort_values('Count', ascending = False)

In both cases, most tract + blockgroup combinations have all four years of data. For ER use:

* BGs 40400-1, 562000-2, 10300-2, and 70500-2 have 3 years of data;
* 563200-1 has 2 years of data;
* 980800-1, 980700-1, 980600-1, 980000-1, 980900-1, 981100-1, 981200-1, and 10300-1 all have 1 year of data

For PC use:

* BG 10300-2 has 3 years of data;
* 563800-4 has 2 years of data;
* 980000-1 has 1 year of data

In [18]:
# Bring in all Allegheny County blockgroups so we can ensure we have all of them
alCoBGs.head()
alCoBGs_subset = alCoBGs[['geoid', 'tract_ce', 'blkgrp_ce']]
alCoBGs_subset.head()
alCoBGs_subset['tract_ce'].nunique()

402

In [19]:
ER_use[ER_use['TractFIPS'] == 981100]
ER_use[ER_use['YearOfStartDate'].isna()]

PC_use[PC_use['YearOfContactDate'].isna()]

Empty DataFrame
Columns: [_id, Geo_FIPS, StateFIPS, CountyFIPS, TractFIPS, BlockgroupFIPS, TotalPopEst2015_19ACS, Age0to17PopEst2015_19ACS, YearOfContactDate, AllPrimaryCarePatientsAge0to17, WellChildVisitsInPastYearAge0to17, WellChildVisitsInPastYearPer100PrimaryCarePatients, AsthmaDiagnosisAge0to17, AsthmaDiagnosisPer100PrimaryCarePatientsAge0to17]
Index: []

In [20]:
# Add placeholder rows for tracts + blockgroups that are missing
# First need to add years 2016-19 to base blockgroup data frame
years = pd.DataFrame({'year': [2016, 2017, 2018, 2019]})
alCoBGs_subset['key'] = 1 # Use key columns to join
years['key'] = 1
base_df = alCoBGs_subset.merge(years, on = 'key').drop(columns = 'key')

# Merge base data frame containing all tracts, blockgroups, and years with the ER use dataset
ER_expanded = base_df.merge(ER_use, how = 'left', left_on = ['tract_ce', 'blkgrp_ce', 'year'], right_on = ['TractFIPS', 'BlockgroupFIPS', 'YearOfStartDate']).sort_values(by = '_id')
ER_expanded.tail(50)
ER_expanded['Geo_FIPS'] = ER_expanded['Geo_FIPS'].fillna(ER_expanded['geoid'])
ER_expanded['TractFIPS'] = ER_expanded['TractFIPS'].fillna(ER_expanded['tract_ce'])
ER_expanded['BlockgroupFIPS'] = ER_expanded['BlockgroupFIPS'].fillna(ER_expanded['blkgrp_ce'])
ER_expanded['YearOfStartDate'] = ER_expanded['YearOfStartDate'].fillna(ER_expanded['year'])
ER_expanded['StateFIPS'] = ER_expanded['StateFIPS'].fillna(42)
ER_expanded['CountyFIPS'] = ER_expanded['CountyFIPS'].fillna(3)
ER_expanded.shape

# Merge base + ER data frame with the PC use dataset
merged_data = ER_expanded.merge(PC_use, how = 'left', left_on = ['tract_ce', 'blkgrp_ce', 'year'], right_on = ['TractFIPS', 'BlockgroupFIPS', 'YearOfContactDate'], suffixes=('_er', '_pc'))
merged_data.head(13)
util_data = merged_data.drop(columns = ['tract_ce', 'blkgrp_ce', 'year', '_id_er', '_id_pc', 'Geo_FIPS_pc', 'StateFIPS_pc', 'CountyFIPS_pc', 'TractFIPS_pc', 'BlockgroupFIPS_pc', 'TotalPopEst2015_19ACS_pc', 'Age0to17PopEst2015_19ACS_pc'])
util_data = util_data.rename(columns = {'Geo_FIPS_er':'Geo_FIPS', 'StateFIPS_er':'StateFIPS', 'CountyFIPS_er':'CountyFIPS', 'TractFIPS_er':'TractFIPS', 'BlockgroupFIPS_er':'BlockgroupFIPS', 'TotalPopEst2015_19ACS_er':'TotalPopEst2015_19ACS'})
util_data.shape

C:\Users\Mahima Batheja\AppData\Local\Temp\ipykernel_37016\1826672544.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  alCoBGs_subset['key'] = 1 # Use key columns to join


(4400, 29)

In [21]:
#merging utilization data for Asthma ER and primary care
util_data = PC_use.merge(ER_use, how = 'left', left_on = ['Geo_FIPS', 'YearOfContactDate'], right_on = ['Geo_FIPS', 'YearOfStartDate'], validate='one_to_one')
#dropping repeat columns
util_data = util_data.drop(["_id_y","StateFIPS_y","CountyFIPS_y","TractFIPS_y","BlockgroupFIPS_y",
                            "TotalPopEst2015_19ACS_y","Age0to17PopEst2015_19ACS_y"],axis="columns")
util_data

_id_x      Geo_FIPS  StateFIPS_x  CountyFIPS_x  TractFIPS_x  \
0         1  420031301002           42             3       130100   
1         2  420031301002           42             3       130100   
2         3  420031301002           42             3       130100   
3         4  420031301002           42             3       130100   
4         5  420031301003           42             3       130100   
...     ...           ...          ...           ...          ...   
4361   4362  420039818001           42             3       981800   
4362   4363  420039822001           42             3       982200   
4363   4364  420039822001           42             3       982200   
4364   4365  420039822001           42             3       982200   
4365   4366  420039822001           42             3       982200   

      BlockgroupFIPS_x  TotalPopEst2015_19ACS_x  Age0to17PopEst2015_19ACS_x  \
0                    2                      518                         207   
1                    2                      518                         207   
2                    2                      518                         207   
3                    2                      518                         207   
4                    3                      489                         134   
...                ...                      ...                         ...   
4361                 1                      207                          47   
4362                 1                     4618                          80   
4363                 1                     4618                          80   
4364                 1                     4618                          80   
4365                 1                     4618                          80   

      YearOfContactDate  AllPrimaryCarePatientsAge0to17  \
0                  2016                              82   
1                  2017                              94   
2                  2018                              89   
3                  2019                              86   
4                  2016                              53   
...                 ...                             ...   
4361               2019                               4   
4362               2016                              16   
4363               2017                              18   
4364               2018                              19   
4365               2019                              16   

      WellChildVisitsInPastYearAge0to17  \
0                                    59   
1                                    58   
2                                    63   
3                                    61   
4                                    34   
...                                 ...   
4361                                  3   
4362                                 13   
4363                                 16   
4364                                 14   
4365                                  8   

      WellChildVisitsInPastYearPer100PrimaryCarePatients  \
0                                                  72.0    
1                                                  61.7    
2                                                  70.8    
3                                                  70.9    
4                                                  64.2    
...                                                 ...    
4361                                               75.0    
4362                                               81.3    
4363                                               88.9    
4364                                               73.7    
4365                                               50.0    

      AsthmaDiagnosisAge0to17  \
0                        17.0   
1                        19.0   
2                        19.0   
3                        17.0   
4                        18.0   
...                       ...   
4361                      2.0   
4362             

In [22]:
util_data.shape

(4366, 29)

In [23]:
#Joining rows based on CTs
#Dropping Per100 columns
util_data = drop_columns_with_keyword(util_data,'Per100')
util_data['Geo_FIPS'] = util_data['Geo_FIPS'].apply(str)
util_data['Geo_FIPS'] = util_data['Geo_FIPS'].str[:-1]
#Changing Geo_FIPS to contain only Census Tract identifier
util_data['Geo_FIPS'] = util_data['Geo_FIPS'].apply(lambda x: int(x[-6:]))
util_data = util_data.drop(columns=['_id_x', 'StateFIPS_x', 'CountyFIPS_x', 'TractFIPS_x', 'BlockgroupFIPS_x'])
util_data = util_data.groupby(['Geo_FIPS', 'YearOfContactDate'], as_index=False).sum()
util_data.isna().sum()

Geo_FIPS                                                        0
YearOfContactDate                                               0
TotalPopEst2015_19ACS_x                                         0
Age0to17PopEst2015_19ACS_x                                      0
AllPrimaryCarePatientsAge0to17                                  0
WellChildVisitsInPastYearAge0to17                               0
AsthmaDiagnosisAge0to17                                         0
YearOfStartDate                                                 0
UnderAge1PopEst2015_19ACS                                       0
Unique0to17WithED_Visit                                         0
NumberED_VisitsAge0to17                                         0
NumberLowAcuityED_VisitsAge0to17                                0
NumberAsthmaRelatedED_Visits                                    0
NumberED_VisitsByChildrenUnder1YearOld                          0
NumberOfInjuryRelatedVisits                                     0
NumberAcut

In [24]:
util_data

Geo_FIPS  YearOfContactDate  TotalPopEst2015_19ACS_x  \
0        10300               2016                     4147   
1        10300               2017                     4147   
2        10300               2018                     4147   
3        10300               2019                      728   
4        20100               2016                     5490   
...        ...                ...                      ...   
1580    981800               2019                      207   
1581    982200               2016                     4618   
1582    982200               2017                     4618   
1583    982200               2018                     4618   
1584    982200               2019                     4618   

      Age0to17PopEst2015_19ACS_x  AllPrimaryCarePatientsAge0to17  \
0                            155                              50   
1                            155                              41   
2                            155                              44   
3                            122                              47   
4                            201                              71   
...                          ...                             ...   
1580                          47                               4   
1581                          80                              16   
1582                          80                              18   
1583                          80                              19   
1584                          80                              16   

      WellChildVisitsInPastYearAge0to17  AsthmaDiagnosisAge0to17  \
0                                    33                      9.0   
1                                    24                      5.0   
2                                    34                      5.0   
3                                    36                      5.0   
4                                    44                      7.0   
...                                 ...                      ...   
1580                                  3                      2.0   
1581                                 13                      1.0   
1582                                 16                      1.0   
1583                                 14                      1.0   
1584                                  8                      1.0   

      YearOfStartDate  UnderAge1PopEst2015_19ACS  Unique0to17WithED_Visit  \
0                 0.0                        0.0                      0.0   
1                 0.0                        0.0                      0.0   
2                 0.0                        0.0                      0.0   
3                 0.0                        0.0                      0.0   
4                 0.0                        0.0                      0.0   
...               ...                        ...                      ...   
1580           2019.0                        0.0                      4.0   
1581           2016.0                        0.0                      6.0   
1582           2017.0                        0.0                     10.0   
1583           2018.0                        0.0                      7.0   
1584           2019.0                        0.0                      8.0   

      NumberED_VisitsAge0to17  NumberLowAcuityED_VisitsAge0to17  \
0                         0.0                               0.0   
1                         0.0                               0.0   
2                         0.0                               0.0   
3                         0.0                               0.0   
4                         0.0                               0.0   
...                       ...                               ...   
1580                      8.0                               2.0   
1581                     10.0                               2.0   
1582                     10.0                               2.0   
1583                      9.0   

### Race Data

In [25]:
#Census data has to be loaded in for each year in 2016-19
#Census indicates missing values using -
demo2016 = pd.read_csv('our-data/ACSDP5Y2016.DP05-Data.csv', header = 1, na_values='-')
#Adding in timeperiod indicator
demo2016['Year'] = [2016]*len(demo2016)
#Filtering out data that does not pertain to Census Tract identifiers, Median Age,
#or estimates for number of individuals from each racial group
demo2016 = demo2016[['Geography', 'Year',
          'Estimate!!SEX AND AGE!!Median age (years)',
          'Estimate!!RACE!!Total population',
          'Estimate!!RACE!!One race!!White',
          'Estimate!!RACE!!One race!!Black or African American',
          'Estimate!!RACE!!One race!!Asian',
          'Estimate!!RACE!!One race!!American Indian and Alaska Native',
          'Estimate!!RACE!!One race!!Native Hawaiian and Other Pacific Islander',
          'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)',
          'Estimate!!RACE!!One race!!Some other race',
          'Estimate!!RACE!!Two or more races']]
#Renaming columns to be more easily legible
demo2016 = demo2016.rename(columns = {'Estimate!!SEX AND AGE!!Median age (years)':'Median Age',
                           'Estimate!!RACE!!Total population':'Total Population',
                           'Estimate!!RACE!!One race!!White':'White Population',
                           'Estimate!!RACE!!One race!!Black or African American':'Black or African American Population',
                           'Estimate!!RACE!!One race!!Asian':'Asian Population',
                           'Estimate!!RACE!!One race!!American Indian and Alaska Native': 'American Indian and Alaska Native Population',
                           'Estimate!!RACE!!One race!!Native Hawaiian and Other Pacific Islander':'Native Hawaiian and Other Pacific Islander Population',
                           'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)':'Hispanic or Latino Population',
                           'Estimate!!RACE!!One race!!Some other race':'Some Other Race Population',
                           'Estimate!!RACE!!Two or more races':'Two or More Races Population'})

#Process is repeated for 2017, 2018, and 2019 but column names in original dataset change in these years
demo2017 = pd.read_csv('our-data/ACSDP5Y2017.DP05-Data.csv', header = 1, na_values='-')
demo2017['Year'] = [2017]*len(demo2017)
#Filtering reflects different name for columns. The real world value that is being estimated stays constant despite name change.
demo2017 = demo2017[['Geography', 'Year',
          'Estimate!!SEX AND AGE!!Total population!!Median age (years)',
          'Estimate!!RACE!!Total population',
          'Estimate!!RACE!!Total population!!One race!!White',
          'Estimate!!RACE!!Total population!!One race!!Black or African American',
          'Estimate!!RACE!!Total population!!One race!!Asian',
          'Estimate!!RACE!!Total population!!One race!!American Indian and Alaska Native',
          'Estimate!!RACE!!Total population!!One race!!Native Hawaiian and Other Pacific Islander',
          'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)',
          'Estimate!!RACE!!Total population!!One race!!Some other race',
          'Estimate!!RACE!!Total population!!Two or more races']]
demo2017 = demo2017.rename(columns = {'Estimate!!SEX AND AGE!!Total population!!Median age (years)':'Median Age',
                           'Estimate!!RACE!!Total population':'Total Population',
                           'Estimate!!RACE!!Total population!!One race!!White':'White Population',
                           'Estimate!!RACE!!Total population!!One race!!Black or African American':'Black or African American Population',
                           'Estimate!!RACE!!Total population!!One race!!Asian':'Asian Population',
                           'Estimate!!RACE!!Total population!!One race!!American Indian and Alaska Native': 'American Indian and Alaska Native Population',
                           'Estimate!!RACE!!Total population!!One race!!Native Hawaiian and Other Pacific Islander':'Native Hawaiian and Other Pacific Islander Population',
                           'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)':'Hispanic or Latino Population',
                           'Estimate!!RACE!!Total population!!One race!!Some other race':'Some Other Race Population',
                           'Estimate!!RACE!!Total population!!Two or more races':'Two or More Races Population'})

demo2018 = pd.read_csv('our-data/ACSDP5Y2018.DP05-Data.csv', header = 1, na_values='-')
demo2018['Year'] = [2018]*len(demo2018)
demo2018 = demo2018[['Geography', 'Year',
          'Estimate!!SEX AND AGE!!Total population!!Median age (years)',
          'Estimate!!RACE!!Total population',
          'Estimate!!RACE!!Total population!!One race!!White',
          'Estimate!!RACE!!Total population!!One race!!Black or African American',
          'Estimate!!RACE!!Total population!!One race!!Asian',
          'Estimate!!RACE!!Total population!!One race!!American Indian and Alaska Native',
          'Estimate!!RACE!!Total population!!One race!!Native Hawaiian and Other Pacific Islander',
          'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)',
          'Estimate!!RACE!!Total population!!One race!!Some other race',
          'Estimate!!RACE!!Total population!!Two or more races']]
demo2018 = demo2018.rename(columns = {'Estimate!!SEX AND AGE!!Total population!!Median age (years)':'Median Age',
                           'Estimate!!RACE!!Total population':'Total Population',
                           'Estimate!!RACE!!Total population!!One race!!White':'White Population',
                           'Estimate!!RACE!!Total population!!One race!!Black or African American':'Black or African American Population',
                           'Estimate!!RACE!!Total population!!One race!!Asian':'Asian Population',
                           'Estimate!!RACE!!Total population!!One race!!American Indian and Alaska Native': 'American Indian and Alaska Native Population',
                           'Estimate!!RACE!!Total population!!One race!!Native Hawaiian and Other Pacific Islander':'Native Hawaiian and Other Pacific Islander Population',
                           'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)':'Hispanic or Latino Population',
                           'Estimate!!RACE!!Total population!!One race!!Some other race':'Some Other Race Population',
                           'Estimate!!RACE!!Total population!!Two or more races':'Two or More Races Population'})

demo2019 = pd.read_csv('our-data/ACSDP5Y2019.DP05-Data.csv', header = 1, na_values='-')
demo2019['Year'] = [2019]*len(demo2019)
demo2019 = demo2019[['Geography', 'Year',
          'Estimate!!SEX AND AGE!!Total population!!Median age (years)',
          'Estimate!!RACE!!Total population',
          'Estimate!!RACE!!Total population!!One race!!White',
          'Estimate!!RACE!!Total population!!One race!!Black or African American',
          'Estimate!!RACE!!Total population!!One race!!Asian',
          'Estimate!!RACE!!Total population!!One race!!American Indian and Alaska Native',
          'Estimate!!RACE!!Total population!!One race!!Native Hawaiian and Other Pacific Islander',
          'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)',
          'Estimate!!RACE!!Total population!!One race!!Some other race',
          'Estimate!!RACE!!Total population!!Two or more races']]
demo2019 = demo2019.rename(columns = {'Estimate!!SEX AND AGE!!Total population!!Median age (years)':'Median Age',
                           'Estimate!!RACE!!Total population':'Total Population',
                           'Estimate!!RACE!!Total population!!One race!!White':'White Population',
                           'Estimate!!RACE!!Total population!!One race!!Black or African American':'Black or African American Population',
                           'Estimate!!RACE!!Total population!!One race!!Asian':'Asian Population',
                           'Estimate!!RACE!!Total population!!One race!!American Indian and Alaska Native': 'American Indian and Alaska Native Population',
                           'Estimate!!RACE!!Total population!!One race!!Native Hawaiian and Other Pacific Islander':'Native Hawaiian and Other Pacific Islander Population',
                           'Estimate!!HISPANIC OR LATINO AND RACE!!Total population!!Hispanic or Latino (of any race)':'Hispanic or Latino Population',
                           'Estimate!!RACE!!Total population!!One race!!Some other race':'Some Other Race Population',
                           'Estimate!!RACE!!Total population!!Two or more races':'Two or More Races Population'})

#Joining data for all four years to create one dataframe
demo_data = pd.concat([demo2016, demo2017, demo2018, demo2019], axis = 0)
demo_data

Geography  Year  Median Age  Total Population  \
0    1400000US42003010300  2016        21.6              6491   
1    1400000US42003020100  2016        28.9              4212   
2    1400000US42003020300  2016        34.3               821   
3    1400000US42003030500  2016        40.3              2292   
4    1400000US42003040200  2016        24.0              1609   
..                    ...   ...         ...               ...   
397  1400000US42003981000  2019         NaN                 0   
398  1400000US42003981100  2019         NaN                 0   
399  1400000US42003981200  2019         NaN                 0   
400  1400000US42003981800  2019        19.4               207   
401  1400000US42003982200  2019        19.3              4618   

     White Population  Black or African American Population  Asian Population  \
0                4433                                  1573               220   
1                3366                                   347               356   
2                 691                                    44                68   
3                 198                                  1777                25   
4                 571                                   807               153   
..                ...                                   ...               ...   
397                 0                                     0                 0   
398                 0                                     0                 0   
399                 0                                     0                 0   
400               113                                    55                17   
401              3844                                   217               420   

     American Indian and Alaska Native Population  \
0                                              24   
1                                              29   
2                                               0   
3                                              20   
4                                               0   
..                                            ...   
397                                             0   
398                                             0   
399                                             0   
400                                             0   
401                                             0   

     Native Hawaiian and Other Pacific Islander Population  \
0                                                    0       
1                                                    0       
2                                                    0       
3                                                    0       
4                                                   19       
..                                                 ...       
397                                                  0       
398                                                  0       
399                                                  0       
400                                                  0       
401                                                  0       

     Hispanic or Latino Population  Some Other Race Population  \
0                              227                           7   
1                              111                           0   
2                               20                           0   
3                              216                          84   
4                               18                           0   
..                             ...                         ...   
397                              0                           0   
398                              0                           0   
399                              0                           0   
400                             17                          10   
401                            117                          16   

     Two or More Races Population  
0                             234  
1              

In [26]:
#There are values for all 402 census tracts 35 rows are missing values for Median Age
demo_data.isna().sum()

Geography                                                 0
Year                                                      0
Median Age                                               35
Total Population                                          0
White Population                                          0
Black or African American Population                      0
Asian Population                                          0
American Indian and Alaska Native Population              0
Native Hawaiian and Other Pacific Islander Population     0
Hispanic or Latino Population                             0
Some Other Race Population                                0
Two or More Races Population                              0
dtype: int64

In [27]:
#Missing values are in census tract with either no population or very low population
demo_data[demo_data['Median Age'].isna()]

Geography  Year  Median Age  Total Population  \
390  1400000US42003980300  2016         NaN                 0   
391  1400000US42003980400  2016         NaN                 0   
393  1400000US42003980600  2016         NaN                 4   
395  1400000US42003980800  2016         NaN                 0   
397  1400000US42003981000  2016         NaN                 0   
398  1400000US42003981100  2016         NaN                 5   
399  1400000US42003981200  2016         NaN                 0   
388  1400000US42003980000  2017         NaN                 0   
390  1400000US42003980300  2017         NaN                 0   
391  1400000US42003980400  2017         NaN                 0   
393  1400000US42003980600  2017         NaN                 3   
394  1400000US42003980700  2017         NaN                 4   
395  1400000US42003980800  2017         NaN                 0   
397  1400000US42003981000  2017         NaN                 0   
398  1400000US42003981100  2017         NaN                 4   
399  1400000US42003981200  2017         NaN                 0   
388  1400000US42003980000  2018         NaN                 0   
389  1400000US42003980100  2018         NaN                13   
390  1400000US42003980300  2018         NaN                 0   
391  1400000US42003980400  2018         NaN                 0   
392  1400000US42003980500  2018         NaN                 9   
394  1400000US42003980700  2018         NaN                 4   
395  1400000US42003980800  2018         NaN                 0   
397  1400000US42003981000  2018         NaN                 0   
398  1400000US42003981100  2018         NaN                 0   
399  1400000US42003981200  2018         NaN                 0   
388  1400000US42003980000  2019         NaN                 0   
389  1400000US42003980100  2019         NaN                 0   
390  1400000US42003980300  2019         NaN                 0   
391  1400000US42003980400  2019         NaN                 0   
394  1400000US42003980700  2019         NaN                 0   
395  1400000US42003980800  2019         NaN                 0   
397  1400000US42003981000  2019         NaN                 0   
398  1400000US42003981100  2019         NaN                 0   
399  1400000US42003981200  2019         NaN                 0   

     White Population  Black or African American Population  Asian Population  \
390                 0                                     0                 0   
391                 0                                     0                 0   
393                 4                                     0                 0   
395                 0                                     0                 0   
397                 0                                     0                 0   
398                 5                                     0                 0   
399                 0                                     0                 0   
388                 0                                     0                 0   
390                 0                                     0                 0   
391                 0                                     0                 0   
393                 3                                     0                 0   
394                 4                                     0                 0   
395                 0                                     0                 0   
397                 0                                     0                 0   
398                 4                                     0                 0   
399                 0                                     0                 0   
388                 0                                     0                 0   
389                13                                     0                 0   
390                 0                                     0                 0   
391                 0                                  

### Household Income Data

In [28]:
#Census data has to be loaded in for each year in 2016-19
#Census indicates missing values using -
income2016 = pd.read_csv('our-data/ACSST5Y2016.S1903-Data.csv', header = 1, na_values='-')
#Adding time period indicator for each row
income2016['Year'] = [2016]*len(income2016)
#Filtering out columns that do not contain row identifier, number of households, or median income
income2016 = income2016[['Geography', 'Year', 'Total!!Estimate!!Households',
                         'Median income (dollars)!!Estimate!!Households']]
#Renaming columns to be more legible
income2016 = income2016.rename(columns={'Total!!Estimate!!Households':'Households',
                                        'Median income (dollars)!!Estimate!!Households':'Median Income'})

#Process is repeated for 2017, 2018, and 2019 but column names in original dataset change in these years
income2017 = pd.read_csv('our-data/ACSST5Y2017.S1903-Data.csv', header = 1, na_values='-')
income2017['Year'] = [2017]*len(income2017)
#Although names of columns change, same values are still being estimated
income2017 = income2017[['Geography', 'Year',
  'Estimate!!Number!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households',
  'Estimate!!Median income (dollars)!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households']]
income2017 = income2017.rename(columns={'Estimate!!Number!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households':'Households',
                           'Estimate!!Median income (dollars)!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households':'Median Income'})

income2018 = pd.read_csv('our-data/ACSST5Y2018.S1903-Data.csv', header = 1, na_values='-')
income2018['Year'] = [2018]*len(income2018)
income2018 = income2018[['Geography', 'Year',
  'Estimate!!Number!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households',
  'Estimate!!Median income (dollars)!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households']]
income2018 = income2018.rename(columns={'Estimate!!Number!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households':'Households',
                           'Estimate!!Median income (dollars)!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households':'Median Income'})

income2019 = pd.read_csv('our-data/ACSST5Y2019.S1903-Data.csv', header = 1, na_values='-')
income2019['Year'] = [2019]*len(income2019)
income2019 = income2019[['Geography', 'Year',
  'Estimate!!Number!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households',
  'Estimate!!Median income (dollars)!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households']]
income2019 = income2019.rename(columns={'Estimate!!Number!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households':'Households',
                           'Estimate!!Median income (dollars)!!HOUSEHOLD INCOME BY RACE AND HISPANIC OR LATINO ORIGIN OF HOUSEHOLDER!!Households':'Median Income'})

#Joining data for all four years to create one dataframe
income_data=pd.concat([income2016,income2017,income2018,income2019])
income_data

Geography  Year  Households  Median Income
0    1400000US42003010300  2016         281        22292.0
1    1400000US42003020100  2016        2042        72642.0
2    1400000US42003020300  2016         553        93125.0
3    1400000US42003030500  2016        1117        18730.0
4    1400000US42003040200  2016         611        16250.0
..                    ...   ...         ...            ...
397  1400000US42003981000  2019           0            NaN
398  1400000US42003981100  2019           0            NaN
399  1400000US42003981200  2019           0            NaN
400  1400000US42003981800  2019           0            NaN
401  1400000US42003982200  2019           8            NaN

[1608 rows x 4 columns]

In [29]:
#Missing values for 50 census tracts
income_data.isna().sum()

Geography         0
Year              0
Households        0
Median Income    50
dtype: int64

In [30]:
#All but one of these census tracts has either no households or a very small number or households
income_data[income_data['Median Income'].isna()]

Geography  Year  Households  Median Income
389  1400000US42003980100  2016          17            NaN
390  1400000US42003980300  2016           0            NaN
391  1400000US42003980400  2016           0            NaN
392  1400000US42003980500  2016           8            NaN
393  1400000US42003980600  2016           4            NaN
394  1400000US42003980700  2016          10            NaN
395  1400000US42003980800  2016           0            NaN
396  1400000US42003980900  2016           0            NaN
397  1400000US42003981000  2016           0            NaN
398  1400000US42003981100  2016           5            NaN
399  1400000US42003981200  2016           0            NaN
91   1400000US42003261500  2017         745            NaN
388  1400000US42003980000  2017           0            NaN
390  1400000US42003980300  2017           0            NaN
391  1400000US42003980400  2017           0            NaN
392  1400000US42003980500  2017           8            NaN
393  1400000US42003980600  2017           3            NaN
394  1400000US42003980700  2017           4            NaN
395  1400000US42003980800  2017           0            NaN
396  1400000US42003980900  2017           0            NaN
397  1400000US42003981000  2017           0            NaN
398  1400000US42003981100  2017           4            NaN
399  1400000US42003981200  2017           0            NaN
388  1400000US42003980000  2018           0            NaN
389  1400000US42003980100  2018          13            NaN
390  1400000US42003980300  2018           0            NaN
391  1400000US42003980400  2018           0            NaN
392  1400000US42003980500  2018           9            NaN
393  1400000US42003980600  2018           8            NaN
394  1400000US42003980700  2018           4            NaN
395  1400000US42003980800  2018           0            NaN
396  1400000US42003980900  2018           0            NaN
397  1400000US42003981000  2018           0            NaN
398  1400000US42003981100  2018           0            NaN
399  1400000US42003981200  2018           0            NaN
400  1400000US42003981800  2018           8            NaN
388  1400000US42003980000  2019           0            NaN
389  1400000US42003980100  2019           0            NaN
390  1400000US42003980300  2019           0            NaN
391  1400000US42003980400  2019           0            NaN
392  1400000US42003980500  2019          16            NaN
393  1400000US42003980600  2019           8            NaN
394  1400000US42003980700  2019           0            NaN
395  1400000US42003980800  2019           0            NaN
396  1400000US42003980900  2019           0            NaN
397  1400000US42003981000  2019           0            NaN
398  1400000US42003981100  2019           0            NaN
399  1400000US42003981200  2019           0            NaN
400  1400000US42003981800  2019           0            NaN
401  1400000US42003982200  2019           8            NaN

### Merging the two Census Datasets

In [31]:
census_data = demo_data.merge(income_data, how='inner', on=['Geography', 'Year'], validate='one_to_one')
#Creating indentifier that contains only census tract information
census_data['Geo_FIPS'] = census_data['Geography'].apply(lambda x: int(x[-6:]))
census_data

Geography  Year  Median Age  Total Population  \
0     1400000US42003010300  2016        21.6              6491   
1     1400000US42003020100  2016        28.9              4212   
2     1400000US42003020300  2016        34.3               821   
3     1400000US42003030500  2016        40.3              2292   
4     1400000US42003040200  2016        24.0              1609   
...                    ...   ...         ...               ...   
1603  1400000US42003981000  2019         NaN                 0   
1604  1400000US42003981100  2019         NaN                 0   
1605  1400000US42003981200  2019         NaN                 0   
1606  1400000US42003981800  2019        19.4               207   
1607  1400000US42003982200  2019        19.3              4618   

      White Population  Black or African American Population  \
0                 4433                                  1573   
1                 3366                                   347   
2                  691                                    44   
3                  198                                  1777   
4                  571                                   807   
...                ...                                   ...   
1603                 0                                     0   
1604                 0                                     0   
1605                 0                                     0   
1606               113                                    55   
1607              3844                                   217   

      Asian Population  American Indian and Alaska Native Population  \
0                  220                                            24   
1                  356                                            29   
2                   68                                             0   
3                   25                                            20   
4                  153                                             0   
...                ...                                           ...   
1603                 0                                             0   
1604                 0                                             0   
1605                 0                                             0   
1606                17                                             0   
1607               420                                             0   

      Native Hawaiian and Other Pacific Islander Population  \
0                                                     0       
1                                                     0       
2                                                     0       
3                                                     0       
4                                                    19       
...                                                 ...       
1603                                                  0       
1604                                                  0       
1605                                                  0       
1606                                                  0       
1607                                                  0       

      Hispanic or Latino Population  Some Other Race Population  \
0                               227                           7   
1                               111                           0   
2                                20                           0   
3                               216                          84   
4                                18                           0   
...                             ...                         ...   
1603                              0                           0   
1604                              0                           0   
1605                              0                           0   
1606                             17                          10   
1607                            117                          16   

      Two or More Races Popu

In [32]:
#Same values contain NaNs
census_data.isna().sum()

Geography                                                 0
Year                                                      0
Median Age                                               35
Total Population                                          0
White Population                                          0
Black or African American Population                      0
Asian Population                                          0
American Indian and Alaska Native Population              0
Native Hawaiian and Other Pacific Islander Population     0
Hispanic or Latino Population                             0
Some Other Race Population                                0
Two or More Races Population                              0
Households                                                0
Median Income                                            50
Geo_FIPS                                                  0
dtype: int64

### Emissions Data

Emissions data is weighted on inverse distance from facility producing them. See other notebook for process calculating values.

In [33]:
emissions_data = pd.read_csv('our-data/final_emissions_data.csv', index_col=0)
emissions_data

TRACTCE    Year  1,1,2-Trichloroethane  1,1,2,2-Tetrachloroethane  \
0      50900  2016.0                    0.0                        0.0   
1      70300  2016.0                    0.0                        0.0   
2     120700  2016.0                    0.0                        0.0   
3     140400  2016.0                    0.0                        0.0   
4     180700  2016.0                    0.0                        0.0   
..       ...     ...                    ...                        ...   
397   141000  2019.0                    0.0                        0.0   
398   202300  2019.0                    0.0                        0.0   
399   320600  2019.0                    0.0                        0.0   
400   408001  2019.0                    0.0                        0.0   
401   414101  2019.0                    0.0                        0.0   

     1,2-Ethanediol (Ethylene Glycol)  1,3-Butadiene  1,4-Dichlorobenzene  \
0                            0.000000            0.0                  0.0   
1                            0.011032            0.0                  0.0   
2                            0.026404            0.0                  0.0   
3                            0.021126            0.0                  0.0   
4                            0.000000            0.0                  0.0   
..                                ...            ...                  ...   
397                          0.009801            0.0                  0.0   
398                          0.000000            0.0                  0.0   
399                          0.000000            0.0                  0.0   
400                          0.000000            0.0                  0.0   
401                          0.000000            0.0                  0.0   

     1,4-Dioxane (1,4-Diethyleneoxide)  2,2,4-Trimethylpentane  Acetaldehyde  \
0                                  0.0                0.016377      0.001580   
1                                  0.0                0.017433      0.000826   
2                                  0.0                0.015003      0.000000   
3                                  0.0                0.014855      0.000000   
4                                  0.0                0.000000      0.001626   
..                                 ...                     ...           ...   
397                                0.0                0.000000      0.000000   
398                                0.0                0.000000      0.000000   
399                                0.0                0.000000      0.000000   
400                                0.0                0.000000      0.000000   
401                                0.0                0.000000      0.000000   

     Acrolein  Acrylic Acid  Acrylonitrile   Ammonia  Antimony  Arsenic  \
0         0.0           0.0            0.0  0.847486       0.0      0.0   
1         0.0           0.0            0.0  0.485450       0.0      0.0   
2         0.0           0.0            0.0  0.055593       0.0      0.0   
3         0.0           0.0            0.0  0.245791       0.0      0.0   
4         0.0           0.0            0.0  0.516487       0.0      0.0   
..        ...           ...            ...       ...       ...      ...   
397       0.0           0.0            0.0  3.324278       0.0      0.0   
398       0.0           0.0            0.0  0.050887       0.0      0.0   
399       0.0           0.0            0.0  0.176331       0.0      0.0   
400       0.0           0.0            0.0  0.000000       0.0      0.0   
401       0.0           0.0            0.0  0.000000       0.0      0.0   

      Benzene  Cadmium  Carbon Dioxide  Carbon Disulfide  Carbon Monoxide  \
0    0.072221      0.0    51097.646442          0.000000        22.786588   
1    0.060666      0.0    32090.112989          0.000000        15.331787   
2    0.035833      0.0     8063.066062          0.000000         5.236010   
3    0.047832      0.

### Final Merges

In [34]:
#Merging health care utlization data with census data on race and household income
census_data['Geo_FIPS'] = census_data['Geo_FIPS'].astype(int)
utilxcensus_data = util_data.merge(census_data, how='left', left_on=['Geo_FIPS','YearOfContactDate'], right_on=['Geo_FIPS','Year'])
utilxcensus_data['Geo_FIPS']=utilxcensus_data['Geo_FIPS'].apply(int)
utilxcensus_data

Geo_FIPS  YearOfContactDate  TotalPopEst2015_19ACS_x  \
0        10300               2016                     4147   
1        10300               2017                     4147   
2        10300               2018                     4147   
3        10300               2019                      728   
4        20100               2016                     5490   
...        ...                ...                      ...   
1580    981800               2019                      207   
1581    982200               2016                     4618   
1582    982200               2017                     4618   
1583    982200               2018                     4618   
1584    982200               2019                     4618   

      Age0to17PopEst2015_19ACS_x  AllPrimaryCarePatientsAge0to17  \
0                            155                              50   
1                            155                              41   
2                            155                              44   
3                            122                              47   
4                            201                              71   
...                          ...                             ...   
1580                          47                               4   
1581                          80                              16   
1582                          80                              18   
1583                          80                              19   
1584                          80                              16   

      WellChildVisitsInPastYearAge0to17  AsthmaDiagnosisAge0to17  \
0                                    33                      9.0   
1                                    24                      5.0   
2                                    34                      5.0   
3                                    36                      5.0   
4                                    44                      7.0   
...                                 ...                      ...   
1580                                  3                      2.0   
1581                                 13                      1.0   
1582                                 16                      1.0   
1583                                 14                      1.0   
1584                                  8                      1.0   

      YearOfStartDate  UnderAge1PopEst2015_19ACS  Unique0to17WithED_Visit  \
0                 0.0                        0.0                      0.0   
1                 0.0                        0.0                      0.0   
2                 0.0                        0.0                      0.0   
3                 0.0                        0.0                      0.0   
4                 0.0                        0.0                      0.0   
...               ...                        ...                      ...   
1580           2019.0                        0.0                      4.0   
1581           2016.0                        0.0                      6.0   
1582           2017.0                        0.0                     10.0   
1583           2018.0                        0.0                      7.0   
1584           2019.0                        0.0                      8.0   

      NumberED_VisitsAge0to17  NumberLowAcuityED_VisitsAge0to17  \
0                         0.0                               0.0   
1                         0.0                               0.0   
2                         0.0                               0.0   
3                         0.0                               0.0   
4                         0.0                               0.0   
...                       ...                               ...   
1580                      8.0                               2.0   
1581                     10.0                               2.0   
1582                     10.0                               2.0   
1583                      9.0   

In [35]:
#merged census and utilization data with health condition prevalence data
utilxcensus_data = utilxcensus_data.dropna(subset=['YearOfContactDate'])
utilxcensus_data['YearOfContactDate'] = utilxcensus_data['YearOfContactDate'].astype(int)
prevalence_data['Year']= prevalence_data['Year'].astype(int)
semi_data = utilxcensus_data.merge(prevalence_data, how='right', left_on=['Geo_FIPS', 'Year'], right_on=['Geo_FIPS','Year'])
semi_data

Geo_FIPS  YearOfContactDate  TotalPopEst2015_19ACS_x  \
0        10300             2019.0                    728.0   
1        20100             2019.0                   5490.0   
2        20300             2019.0                   1212.0   
3        30500             2019.0                   2109.0   
4        40200             2019.0                   1576.0   
...        ...                ...                      ...   
1563    564000             2016.0                   6590.0   
1564    564100             2016.0                    883.0   
1565    564200             2016.0                   2091.0   
1566    564400             2016.0                   5874.0   
1567    564500             2016.0                   2862.0   

      Age0to17PopEst2015_19ACS_x  AllPrimaryCarePatientsAge0to17  \
0                          122.0                            47.0   
1                          201.0                            67.0   
2                           41.0                            38.0   
3                          381.0                           280.0   
4                          100.0                            68.0   
...                          ...                             ...   
1563                      1597.0                           511.0   
1564                       181.0                            32.0   
1565                       272.0                           123.0   
1566                       910.0                           408.0   
1567                       590.0                           334.0   

      WellChildVisitsInPastYearAge0to17  AsthmaDiagnosisAge0to17  \
0                                  36.0                      5.0   
1                                  48.0                      5.0   
2                                  39.0                      2.0   
3                                 219.0                     49.0   
4                                  50.0                     17.0   
...                                 ...                      ...   
1563                              428.0                     54.0   
1564                               30.0                      5.0   
1565                              101.0                     19.0   
1566                              341.0                     57.0   
1567                              289.0                     44.0   

      YearOfStartDate  UnderAge1PopEst2015_19ACS  Unique0to17WithED_Visit  \
0                 0.0                        0.0                      0.0   
1                 0.0                        0.0                      0.0   
2                 0.0                        0.0                      0.0   
3                 0.0                        0.0                      0.0   
4                 0.0                        0.0                      0.0   
...               ...                        ...                      ...   
1563           6048.0                       90.6                    155.0   
1564           4032.0                        7.8                     15.0   
1565           6048.0                        6.6                     48.0   
1566          14112.0                       48.2                    188.0   
1567           6048.0                       39.2                     48.0   

      NumberED_VisitsAge0to17  NumberLowAcuityED_VisitsAge0to17  \
0                         0.0                               0.0   
1                         0.0                               0.0   
2                         0.0                               0.0   
3                         0.0                               0.0   
4                         0.0                               0.0   
...                       ...                               ...   
1563                    196.0                              57.0   
1564                     16.0                               5.0   
1565                     70.0                              12.0   
1566                    315.0   

In [36]:
#joining emissions dataset
data = semi_data.merge(emissions_data, how='inner', left_on=['Geo_FIPS', 'YearOfContactDate'], right_on=['TRACTCE','Year'])
data

Geo_FIPS  YearOfContactDate  TotalPopEst2015_19ACS_x  \
0        10300             2019.0                    728.0   
1        20100             2019.0                   5490.0   
2        20300             2019.0                   1212.0   
3        30500             2019.0                   2109.0   
4        40200             2019.0                   1576.0   
...        ...                ...                      ...   
1555    564000             2016.0                   6590.0   
1556    564100             2016.0                    883.0   
1557    564200             2016.0                   2091.0   
1558    564400             2016.0                   5874.0   
1559    564500             2016.0                   2862.0   

      Age0to17PopEst2015_19ACS_x  AllPrimaryCarePatientsAge0to17  \
0                          122.0                            47.0   
1                          201.0                            67.0   
2                           41.0                            38.0   
3                          381.0                           280.0   
4                          100.0                            68.0   
...                          ...                             ...   
1555                      1597.0                           511.0   
1556                       181.0                            32.0   
1557                       272.0                           123.0   
1558                       910.0                           408.0   
1559                       590.0                           334.0   

      WellChildVisitsInPastYearAge0to17  AsthmaDiagnosisAge0to17  \
0                                  36.0                      5.0   
1                                  48.0                      5.0   
2                                  39.0                      2.0   
3                                 219.0                     49.0   
4                                  50.0                     17.0   
...                                 ...                      ...   
1555                              428.0                     54.0   
1556                               30.0                      5.0   
1557                              101.0                     19.0   
1558                              341.0                     57.0   
1559                              289.0                     44.0   

      YearOfStartDate  UnderAge1PopEst2015_19ACS  Unique0to17WithED_Visit  \
0                 0.0                        0.0                      0.0   
1                 0.0                        0.0                      0.0   
2                 0.0                        0.0                      0.0   
3                 0.0                        0.0                      0.0   
4                 0.0                        0.0                      0.0   
...               ...                        ...                      ...   
1555           6048.0                       90.6                    155.0   
1556           4032.0                        7.8                     15.0   
1557           6048.0                        6.6                     48.0   
1558          14112.0                       48.2                    188.0   
1559           6048.0                       39.2                     48.0   

      NumberED_VisitsAge0to17  NumberLowAcuityED_VisitsAge0to17  \
0                         0.0                               0.0   
1                         0.0                               0.0   
2                         0.0                               0.0   
3                         0.0                               0.0   
4                         0.0                               0.0   
...                       ...                               ...   
1555                    196.0                              57.0   
1556                     16.0                               5.0   
1557                     70.0                              12.0   
1558                    315.0   

In [39]:
#Filtering out columns that are not important and then downloading csv
cols = ['Geo_FIPS','YearOfContactDate',
 'TotalPopEst2015_19ACS_x',
 'Age0to17PopEst2015_19ACS_x',
 'AllPrimaryCarePatientsAge0to17',
 'WellChildVisitsInPastYearAge0to17',
 'AsthmaDiagnosisAge0to17',
 'UnderAge1PopEst2015_19ACS',
 'Unique0to17WithED_Visit',
 'NumberED_VisitsAge0to17',
 'NumberLowAcuityED_VisitsAge0to17',
 'NumberAsthmaRelatedED_Visits',
 'NumberED_VisitsByChildrenUnder1YearOld',
 'NumberOfInjuryRelatedVisits',
 'NumberAcuteRespiratoryTractInfectionRelatedED_VisitsAge0to17',
 'Median Age',
 'Total Population',
 'White Population',
 'Black or African American Population',
 'Asian Population',
 'American Indian and Alaska Native Population',
 'Native Hawaiian and Other Pacific Islander Population',
 'Hispanic or Latino Population',
 'Some Other Race Population',
 'Two or More Races Population',
 'Households',
 'Median Income',
 'ACCESS2_CrudePrev',
 'BPHIGH_CrudePrev',
 'BPMED_CrudePrev',
 'CASTHMA_CrudePrev',
 'CHD_CrudePrev',
 'CHECKUP_CrudePrev',
 'CHOLSCREEN_CrudePrev',
 'COPD_CrudePrev',
 'COREM_CrudePrev',
 'CSMOKING_CrudePrev',
 'DIABETES_CrudePrev',
 'HIGHCHOL_CrudePrev',
 'OBESITY_CrudePrev',
 '1,1,2-Trichloroethane',
 '1,1,2,2-Tetrachloroethane',
 '1,2-Ethanediol (Ethylene Glycol)',
 '1,3-Butadiene',
 '1,4-Dichlorobenzene',
 '1,4-Dioxane (1,4-Diethyleneoxide)',
 '2,2,4-Trimethylpentane',
 'Acetaldehyde',
 'Acrolein',
 'Acrylic Acid',
 'Acrylonitrile',
 'Ammonia',
 'Antimony',
 'Arsenic',
 'Benzene',
 'Cadmium',
 'Carbon Dioxide',
 'Carbon Disulfide',
 'Carbon Monoxide',
 'Carbon Tetrachloride',
 'Chlorine',
 'Chlorobenzene',
 'Chloroethene (vinyl chloride)',
 'Chloroform',
 'Chromium',
 'Cobalt',
 'Cresols/Cresylic Acid (Isomers And Mixture)',
 'Cyanides',
 'Dibutylphthalate',
 'Ethyl Chloride (Chloroethane)',
 'Ethylbenzene',
 'Ethylene Dichloride (1,2-Dichloroethane)',
 'Ethylene Oxide',
 'Ethylidene Dichloride (1,1-Dichloroethane)',
 'Formaldehyde',
 'Glycol Ethers',
 'Hexane',
 'Hexavalent Chromium',
 'Hydrochloric Acid',
 'Hydrogen Fluoride (Hydrofluoric Acid)',
 'Hydrogen Sulfide',
 'Isophorone',
 'Lead',
 'm-Xylene',
 'Maleic Anhydride',
 'Manganese',
 'Mercury',
 'Methane',
 'Methanol',
 'Methyl Chloride (Chloromethane)',
 'Methyl Chloroform (1,1,1-Trichloroethane)',
 'Methyl Isobutyl Ketone (4-Methyl-2-Pentanone)',
 'Methyl Methacrylate',
 'Methylene Chloride (Dichloromethane)',
 'Naphthalene',
 'Nickel',
 'Nitrogen Oxides',
 'Nitrous Oxide (N2O)',
 'o-Xylene',
 'Particulate Matter < 10 Microns, Filterable',
 'Particulate Matter < 2.5 Microns, Filterable',
 'Particulate Matter, Condensable',
 'Phenol',
 'Phosphorus',
 'Phthalic Anhydride',
 'Polycyclic Organic Matter',
 'Propionaldehyde',
 'Propylene Oxide',
 'Selenium',
 'Styrene',
 'Sulfur Oxides',
 'Sulfuric Acid',
 'Tetrachloroethylene (Perchloroethylene)',
 'Toluene',
 'Trichloroethylene',
 'Vinylidene Chloride (1,1-Dichloroethylene)',
 'Volatile Organic Compounds',
 'Xylenes (Isomers And Mixture)',
 'Grand Total']
data = data[cols]
data.to_csv('data.csv')

In [41]:
#Loading in Pre1950s Housing values for additional features
pre1950housing = pd.read_csv('our-data/pre1950housing.xls-pre1950housing.csv')
h2016 = pre1950housing.copy()
h2016['Year'] = [2016]*len(h2016)
h2017 = pre1950housing.copy()
h2017['Year'] = [2017]*len(h2017)
h2018 = pre1950housing.copy()
h2018['Year'] = [2018]*len(h2018)
h2019 = pre1950housing.copy()
h2019['Year'] = [2019]*len(h2019)
pre1950 = pd.concat([h2016, h2017, h2018, h2019])
pre1950
test = data.merge(pre1950, how = 'left', left_on=['Geo_FIPS','YearOfContactDate'], right_on=['TRACT','Year'])
test.to_csv('our-data/data_W_1950s.csv')